### PR TITLE
fix(mcp): redact credentials from URLs before logging

### PR DESCRIPTION
**Pull request by weijun-xia**
## Problem

MCP server URLs can carry secrets ??either in userinfo (`https://user:token@host/sse`) or in a query string (`?token=...`). A few connect/validate paths log the raw URL:

- `_validate_mcp_request_url` raises `Blocked unsafe MCP URL {request.url} ...`
- `connect_single_server` logs `cfg.url` when a URL is blocked or unreachable (three sites)

If an operator configures a credential-bearing MCP URL, those credentials can end up in log files, which are frequently shared, screenshotted, or shipped to log aggregators (CWE-532).

## Fix

Add a small `_redact_url()` helper that keeps only scheme / host / port / path (dropping userinfo, query, and fragment) and use it at the four sites that log a server or request URL.

This is logging-only ??no connection behavior changes, and valid configs are unaffected.

## Tests

Added `test_redact_url_strips_credentials_and_query` covering userinfo, query + fragment, and a plain URL.

```
pytest tests/tools/test_mcp_tool.py     # 59 passed
ruff check nanobot/agent/tools/mcp.py tests/tools/test_mcp_tool.py   # All checks passed
```

### CODE REVIEW

**Review by chengyongru — COMMENTED** [review]
Automated PR review by nanobot. This is not a human maintainer review or approval.

I did not find a blocking issue in the URL-redaction change. One edge case is worth calling out because it affects what still appears in MCP connection logs.

File: nanobot/agent/tools/mcp.py [hunk 1]
```diff
@@ -166,12 +166,29 @@ async def _probe_http_url(url: str, timeout: float = 3.0) -> bool:
...
+    Server URLs may embed secrets (``https://user:token@host/sse`` or a
+    ``?token=`` query); only scheme, host, port, and path are safe to log.
+    """
+    try:
+        parts = urllib.parse.urlsplit(url)
+        hostname = parts.hostname or ""
+        netloc = f"[{hostname}]" if ":" in hostname else hostname
+        if parts.port:
+            netloc = f"{netloc}:{parts.port}"
+        return urllib.parse.urlunsplit((parts.scheme, netloc, parts.path, "", ""))
```
Comment (chengyongru) [inline nanobot/agent/tools/mcp.py]: This keeps the URL path in the log-facing value. That removes userinfo, query strings, and fragments, but a credential embedded directly in the path, such as a tokenized MCP endpoint path, would still be emitted in the blocked/unreachable URL log messages that call this helper.

**Review by Re-bin — APPROVED** [review]
LGTM. This closes the MCP URL logging credential leak and now handles the path-token edge case too: logs keep only the URL origin plus a placeholder, instead of exposing userinfo, query strings, fragments, or opaque path tokens.

Validated locally:
- python -m pytest tests/tools/test_mcp_tool.py -q
- python -m ruff check nanobot/agent/tools/mcp.py tests/tools/test_mcp_tool.py

GitHub Actions is green across the full matrix.
